docs/integrations/mlflow.md

```diff
@@ -0,0 +1,250 @@
+with mlflow.start_run() as run:
+    mlflow.log_input(dataset, context="training")
+    mlflow.set_tag("lakefs_repo", repo_id)
+    mlflow.set_tag("lakefs_branch", branch_id) # Log the branch id, to have a friendly lakeFS reference to search the input dataset in 
+
+# Inspect run's dataset
+logged_run = mlflow.get_run(run.info.run_id) # 
+
+# Retrieve the Dataset object
+logged_dataset = logged_run.inputs.dataset_inputs[0].dataset
+
+# View some of the recorded Dataset information
+print(f"Logged dataset name: {logged_dataset.name}")
+print(f"Logged dataset source URI: {logged_dataset.source}")
+```
+
+Output 
+```text
+Dataset name: boat-images
+Dataset source URI: s3://my-repo/3afddad4fef987b4919f5e82f16682c018f59ed2ff003a6a81adf72edaad23c3/gold/train_v2/
+Logged dataset name: boat-images
+Logged dataset source URI: {"path": "s3://my-repo/3afddad4fef987b4919f5e82f16682c018f59ed2ff003a6a81adf72edaad23c3/gold/train_v2/"}
+```
+
+### Reproduce experiment results
+
+To reproduce the results of a specific experiment run in MLflow, it's essential to retrieve the exact dataset and associated
+metadata used during that run. While the MLflow Tracking UI provides a general overview, detailed dataset information 
+and its source are best accessed programmatically.
+
+1. Obtain the Run ID: Navigate to the MLflow UI and copy the Run ID of the experiment you're interested in.
+
+![mlflow run](../assets/img/mlflow_inspect_experiment_run.png)
+
+2. Extract Dataset Information Using MLflow's Python SDK:
+
+```python
+import mlflow
+
+# Inspect run's dataset and tags
+run_id = "c0f8fbb1b63748abaa0a6479115e272c"
+run = mlflow.get_run(run_id) 
+
+# Retrieve the Dataset object
+logged_dataset = run.inputs.dataset_inputs[0].dataset
+
+# View some of the recorded Dataset information
+print(f"Run ID: {run_id} Dataset name: {logged_dataset.name}")
+print(f"Run ID: {run_id} Dataset source URI: {logged_dataset.source}")
+
+# Retrieve run's tags 
+logged_tags = run.data.tags
+print(f"Run ID: {run_id} tags: {logged_tags}")
+```
+
+Output
+```text
+Run ID: c0f8fbb1b63748abaa0a6479115e272c Dataset name: boat-images
+Run ID: c0f8fbb1b63748abaa0a6479115e272c Dataset source URI: {"path": "s3://my-repo/3afddad4fef987b4919f5e82f16682c018f59ed2ff003a6a81adf72edaad23c3/gold/train_v2/"}
+Run ID: c0f8fbb1b63748abaa0a6479115e272c tags: {'lakefs_branch': 'experiment-1', 'lakefs_repo': 'my-repo'}
+```
+
+Notes:
+* The Dataset Source URI provides the location of the dataset at the exact version used in the run. 
+* Run tags, such as 'lakefs_branch' and 'lakefs_repo', offer additional context about the dataset's origin within lakeFS.
+
+### Compare runs input
+
+To determine whether two distinct MLflow runs utilized the same input dataset, you can compare specific attributes of 
+their logged Dataset objects. The source attribute, which contains the versioned dataset's URI, is a common choice for 
+this comparison. Here's an example:
```

Comment:
how the following approach works with how mlflow compare the results? https://mlflow.org/docs/latest/getting-started/quickstart-2/index.html#compare-the-results

Reply:
This is orthogonal to their guide, in my opinion. We let MLflow users drill down into additional run details, which they can use to reproduce experiment results or troubleshoot.  
Would it add clarity if I explained why to compare runs input?

Reply:
talked with tal and discuss the option to use the SDK to compare the actual data or provide more information about the data source from the commit.